runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java

```diff
@@ -120,6 +143,37 @@ void priceLookup(String item)
 		executor.execute(() -> priceLookup(true));
 	}
 
+	public void resizePanel(Dimension parentSize)
+	{
+		this.parentSize = parentSize;
+
+		resultsWrapper.setPreferredSize(calculatePrefSize());
+
+		resultsWrapper.revalidate();
+		resultsWrapper.repaint();
+	}
+
+	/**
+	 * This method calculates the prefered size for the scroll pane.
+	 * I unfortunately cannot make the scroll panel resize automatically so I'll have to
+	 * recalculate everytime the search changes or the client resizes.
+	 *
+	 * If all items fit on screen, the size is set to their heights (58 height + 5 margin) combined,
+	 * if not, the size will be set to the full panel height - 100 pixels (enough room for the search bar and tabs).
+	 *
+	 */
+	private Dimension calculatePrefSize()
+	{
+		int availableHeight = parentSize.height - 100;
+
+		if (ITEMS_LIST.isEmpty() || ITEMS_LIST.size() > (availableHeight / 63))
```

Comment:
This 63 seems to be a magic constant.
You do say in the javadoc above that the list entry height of an item is 58 + 5 (which adds up to 63)

Could you make a static final int that is named for this?
Suppose: `ITEM_LIST_BLOCK_HEIGHT = 58 + 5`  with a comment explaining that the entry height is 58 and the margin is 5. Then use that constant here instead.